### DISCUSSION TITLE
Unable to export html with images in another directory

### DISCUSSION THREAD

**Discussion by TonioGela** [marp-team/marp#122] [Q&A] [answered]
Hello, thanks for Marp, is really awesome.
I'm having some troubles exporting the presentation in HTML in a different output directory than the markdown one.
My directory structure is dead simple:
```
images/foo.png
slides.md
```

when I build the presentation with `marp --html true  slides.md`  the presentation is fine and contains 
`<figure style="background-image:url('images/foo.png');"></figure>` 
in a slide but if I build it using `marp --html true --output ~/Desktop/slides.html slides.md` that figure isn't present in the HTML file, so it's not even fixable moving the `images` directory at destination.

Am I doing something wrong or is it supposed to work this way?

**Comment by TonioGela** [comment]
Plus, I tried adding `--allow-local-files` but it didn't worked

**Comment by yhatt** [comment, marked as answer]
In this case, the root directory of outputted HTML has become `~/Desktop/` so the browser will find out the image from `~/Desktop/images/foo.png`. Marp will not try to modify relative paths for the resource. Respects specified paths in Markdown.

So you must specify resources by the absolute path with `file:` scheme if you want to place HTML in anywhere of local file system.

```markdown
![bg](file:///Users/xxxxxxxx/marp/images/foo.png)
```

If you don't want to replace paths, [(undocumented) `baseUrl` option for Marp CLI >= v1.0.2](https://github.com/marp-team/marp-cli/pull/343) might be useful to set the assumed file path for relative path resolution. It cannot set through CLI options but can set via a config file.

```yaml
# .marprc.yml
# Set "baseUrl" to the absolute path for an original Markdown
baseUrl: file:///Users/xxxxxxx/marp/slides.md
```

```bash
marp -c .marprc.yml -o ~/Desktop/slides.html slides.md
```

**Reply by TonioGela** [reply]
Thanks for the details. At this point which is, in your opinion, the best way to automatize a deploy w/o npm but using simply the command line utility? I planned to move the HTML in a `build` directory, calling the file index.html and serving it, but it doesn't seem easily solvable. Maybe `baseUrl` can help, but even setting it, the problem is that the images won't get copied in the build directory, so they won't be served as static files.

**Reply by TonioGela** [reply]
Btw the original question can be considered solved now, thanks.

**Reply by yhatt** [reply]
Marp is only providing Markdown conversion and don't do extra things. It depends to you how manage resources. I think simply just copying images directory into build directory should be sufficient.

```bash
marp -o ~/build/slides.html slides.md && cp -r images ~/build
```

This approach is adopted in [our Marp CLI example for serving HTML](https://github.com/yhatt/marp-cli-example/blob/1fe4e0ce0cf8f411b80cef7ecc6332b9dcc0e596/package.json#L9) too.

**Reply by TonioGela** [reply]
Thanks, what I did was using another tool that worked out of the box, [monolith](https://github.com/Y2Z/monolith)
